scripts/caclmgrd

```diff
@@ -809,6 +811,9 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
         updates were received during the delay window, at which point it will update
         iptables using the current ACL rules.
         """
+        # ConfigDBConnector is not multi thread safe. In child thread, we use another new DB connector.
+        new_config_db_connector = swsscommon.ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
```

Comment:
>new_config_db_connector [](http://example.com/codeflow?start=8&length=23)

ConfigDBConnector instance holds some native resources, python need to call `close` function to release the resource. even better, use python `with statement` as syntax suger. ref: [https://stackoverflow.com/a/46137552/2514803](https://stackoverflow.com/a/46137552/2514803)

Reply:
@qiluo-msft  there is no `__enter__` function for `ConfigDBConnector`, that's why we can't use with statement here.

```
Jun  7 04:05:27.232731 str-e1031-acs-1 INFO caclmgrd[41190]: ACL change detected for namespace ''
Jun  7 04:05:27.233576 str-e1031-acs-1 INFO caclmgrd[41190]: Spawning ACL update thread for namepsace '' ...
Jun  7 04:05:27.235580 str-e1031-acs-1 INFO caclmgrd[41190]: Exception in thread Thread-1:
Jun  7 04:05:27.236453 str-e1031-acs-1 INFO caclmgrd[41190]: Traceback (most recent call last):
Jun  7 04:05:27.237430 str-e1031-acs-1 INFO caclmgrd[41190]:   File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
Jun  7 04:05:27.249183 str-e1031-acs-1 INFO caclmgrd[41190]:     self.run()
Jun  7 04:05:27.249990 str-e1031-acs-1 INFO caclmgrd[41190]:   File "/usr/lib/python3.9/threading.py", line 892, in run
Jun  7 04:05:27.254364 str-e1031-acs-1 INFO caclmgrd[41190]:     self._target(*self._args, **self._kwargs)
Jun  7 04:05:27.255195 str-e1031-acs-1 INFO caclmgrd[41190]:   File "/usr/local/bin/caclmgrd", line 815, in check_and_update_control_plane_acls
Jun  7 04:05:27.256819 str-e1031-acs-1 INFO caclmgrd[41190]:     with swsscommon.ConfigDBConnector(use_unix_socket_path=True, namespace=namespace) as new_config_db_connector:
Jun  7 04:05:27.257733 str-e1031-acs-1 INFO caclmgrd[41190]: AttributeError: __enter__

```


But I agree with you we need to close or delete the connector when child thread's job is finished.

There is no `close` function in class  `ConfigDBConnector_Native`, I didn't find any example to close connected connectors.


class ConfigDBConnector_Native(SonicV2Connector_Native):
    thisown = property(lambda x: x.this.own(), lambda x, v: x.this.own(v), doc="The membership flag")
    __repr__ = _swig_repr
    INIT_INDICATOR = _swsscommon.ConfigDBConnector_Native_INIT_INDICATOR

    def __init__(self, *args, **kwargs):
        _swsscommon.ConfigDBConnector_Native_swiginit(self, _swsscommon.new_ConfigDBConnector_Native(*args, **kwargs))

    def db_connect(self, db_name, wait_for_init=False, retry_on=False):
        return _swsscommon.ConfigDBConnector_Native_db_connect(self, db_name, wait_for_init, retry_on)

    def connect(self, wait_for_init=True, retry_on=False):
        return _swsscommon.ConfigDBConnector_Native_connect(self, wait_for_init, retry_on)

    def set_entry(self, table, key, data):
        return _swsscommon.ConfigDBConnector_Native_set_entry(self, table, key, data)

    def mod_entry(self, table, key, data):
        return _swsscommon.ConfigDBConnector_Native_mod_entry(self, table, key, data)

    def get_entry(self, table, key):
        return _swsscommon.ConfigDBConnector_Native_get_entry(self, table, key)

    def get_keys(self, table, split=True):
        return _swsscommon.ConfigDBConnector_Native_get_keys(self, table, split)

    def get_table(self, table):
        return _swsscommon.ConfigDBConnector_Native_get_table(self, table)

    def delete_table(self, table):
        return _swsscommon.ConfigDBConnector_Native_delete_table(self, table)

    def mod_config(self, data):
        return _swsscommon.ConfigDBConnector_Native_mod_config(self, data)

    def get_config(self):
        return _swsscommon.ConfigDBConnector_Native_get_config(self)

    def getKeySeparator(self):
        return _swsscommon.ConfigDBConnector_Native_getKeySeparator(self)

    def getTableNameSeparator(self):
        return _swsscommon.ConfigDBConnector_Native_getTableNameSeparator(self)

    def getDbName(self):
        return _swsscommon.ConfigDBConnector_Native_getDbName(self)
    __swig_destroy__ = _swsscommon.delete_ConfigDBConnector_Native

Reply:
@liuh-80 Could you check above comment? Seems a feature missing in classes.

@ZhaohuiS Could you use a `try-finally` to cover all the return path? for example, an exception throw will break the loop.

Reply:
I checked swss-common and feature do missing, I will create a task to track this issue.